x/staking/types/keys.go

```diff
@@ -16,3 +25,272 @@ const (
+
+	// Last* values are constant during a block.
+	LastValidatorPowerKey = []byte{0x11} // prefix for each key to a validator index, for bonded validators
+	LastTotalPowerKey     = []byte{0x12} // prefix for the total power
+
+	ValidatorsKey             = []byte{0x21} // prefix for each key to a validator
+	ValidatorsByConsAddrKey   = []byte{0x22} // prefix for each key to a validator index, by pubkey
+	ValidatorsByPowerIndexKey = []byte{0x23} // prefix for each key to a validator index, sorted by power
+
+	DelegationKey                    = []byte{0x31} // key for a delegation
+	UnbondingDelegationKey           = []byte{0x32} // key for an unbonding-delegation
+	UnbondingDelegationByValIndexKey = []byte{0x33} // prefix for each key for an unbonding-delegation, by validator operator
+	RedelegationKey                  = []byte{0x34} // key for a redelegation
+	RedelegationByValSrcIndexKey     = []byte{0x35} // prefix for each key for an redelegation, by source validator operator
+	RedelegationByValDstIndexKey     = []byte{0x36} // prefix for each key for an redelegation, by destination validator operator
+
+	UnbondingQueueKey    = []byte{0x41} // prefix for the timestamps in unbonding queue
+	RedelegationQueueKey = []byte{0x42} // prefix for the timestamps in redelegations queue
+	ValidatorQueueKey    = []byte{0x43} // prefix for the timestamps in validator queue
+)
+
+// gets the key for the validator with address
+// VALUE: staking/Validator
+func GetValidatorKey(operatorAddr sdk.ValAddress) []byte {
+	return append(ValidatorsKey, operatorAddr.Bytes()...)
+}
+
+// gets the key for the validator with pubkey
+// VALUE: validator operator address ([]byte)
+func GetValidatorByConsAddrKey(addr sdk.ConsAddress) []byte {
+	return append(ValidatorsByConsAddrKey, addr.Bytes()...)
+}
+
+// Get the validator operator address from LastValidatorPowerKey
+func AddressFromLastValidatorPowerKey(key []byte) []byte {
+	return key[1:] // remove prefix bytes
+}
+
+// get the validator by power index.
+// Power index is the key used in the power-store, and represents the relative
+// power ranking of the validator.
+// VALUE: validator operator address ([]byte)
+func GetValidatorsByPowerIndexKey(validator Validator) []byte {
+	// NOTE the address doesn't need to be stored because counter bytes must always be different
+	return getValidatorPowerRank(validator)
+}
+
+// get the bonded validator index key for an operator address
+func GetLastValidatorPowerKey(operator sdk.ValAddress) []byte {
+	return append(LastValidatorPowerKey, operator...)
+}
+
+// get the power ranking of a validator
+// NOTE the larger values are of higher value
+// nolint: unparam
+func getValidatorPowerRank(validator Validator) []byte {
+
+	consensusPower := sdk.TokensToConsensusPower(validator.Tokens)
+	consensusPowerBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(consensusPowerBytes[:], uint64(consensusPower))
+
+	powerBytes := consensusPowerBytes
```

Comment:
consolidated `keys.go` with `key.go` - but these previous 5 lines are the only code changes in this file